### PR TITLE
doc/md: documenting inspect for go api

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: doc/md/go-api/inspection.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+func TestInspect(t *testing.T) {
+	// ... skipping driver creation
+    ctx := context.Background()
+	
+	// Create an example table.
+    _, err = db.ExecContext(ctx, "create table example ( id int not null );")
+    if err != nil {
+    log.Fatalf("failed creating example table: %s", err)
+    }
+
```
Comment (masseelch) [inline doc/md/go-api/inspection.md:57]:
```suggestion
```

File: doc/md/go-api/inspection.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  InspectSchema(ctx context.Context, name string, opts *InspectOptions) (*Schema, error)
+  
+  // InspectTable returns the table description by its name. A NotExistError
+  // error is returned if the table does not exists in the database.
+  InspectTable(ctx context.Context, name string, opts *InspectTableOptions) (*Table, error)
+  
+  // InspectRealm returns the description of the connected database.
+  InspectRealm(ctx context.Context, opts *InspectRealmOption) (*Realm, error)
+}
+```
```
Comment (a8m) [inline doc/md/go-api/inspection.md:32]: Add link to godoc. I'm going to drop the inspect-table one and we don't want to maintain it twice.

File: doc/md/go-api/inspection.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    ctx := context.Background()
+	
+	// Create an example table.
+    _, err = db.ExecContext(ctx, "create table example ( id int not null );")
+    if err != nil {
+    log.Fatalf("failed creating example table: %s", err)
+    }
+	
+	// Inspect a table named "example" in the "main" (SQLite default) schema.
+	tbl, err := driver.InspectTable(ctx, "example", &schema.InspectTableOptions{
```
Comment (a8m) [inline doc/md/go-api/inspection.md:59]: see comment above

File: doc/website/sidebars.js [hunk 4]
```diff
@@ -43,6 +43,7 @@ module.exports = {
             label: 'Go API',
             items: [
                 {type: 'doc', id: 'go-api/intro', label: 'Introduction'},
+                {type: 'doc', id: 'go-api/inspection', label: 'Inspecting Schemas'},
```
Comment (a8m) [inline doc/website/sidebars.js:46]:
```suggestion
                {type: 'doc', id: 'go-api/inspect', label: 'Inspecting Schemas'},
```
maybe inspect instead

File: doc/md/go-api/inspection.md (hunk 2 above)
Comment (rotemtam) [reply doc/md/go-api/inspection.md:32]: See L12 for godoc link :-)

File: doc/md/go-api/inspect.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+// drivers for inspecting multiple tables.
+type Inspector interface {
+  // InspectSchema returns the schema description by its name. An empty name means the
+  // "attached schema" (e.g. SCHEMA() in MySQL or CURRENT_SCHEMA() in PostgreSQL).
+  // A NotExistError error is returned if the schema does not exists in the database.
+  InspectSchema(ctx context.Context, name string, opts *InspectOptions) (*Schema, error)
+  
+  // InspectTable returns the table description by its name. A NotExistError
+  // error is returned if the table does not exists in the database.
+  InspectTable(ctx context.Context, name string, opts *InspectTableOptions) (*Table, error)
```
Comment (a8m) [inline doc/md/go-api/inspect.md:27]: remove

**Pull request merged**